itests/eth_api_test.go

```diff
@@ -0,0 +1,35 @@
+package itests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/chain/wallet/key"
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+func TestEthAddressToFilecoinAddress(t *testing.T) {
+	client, _, _ := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	secpKey, err := key.GenerateKey(types.KTDelegated)
+	require.NoError(t, err)
+
+	filecoinAddr, err := client.WalletImport(ctx, &secpKey.KeyInfo)
+	require.NoError(t, err)
+
+	ethAddr, err := ethtypes.EthAddressFromFilecoinAddress(filecoinAddr)
+	require.NoError(t, err)
+
+	apiFilAddr, err := client.EthAddressToFilecoinAddress(ctx, ethAddr)
+	require.NoError(t, err)
+
+	require.Equal(t, filecoinAddr, apiFilAddr)
```

Comment:
Could you enhance this test or add another test, to test the conversion for non delegated addresses?